src/host.cpp

```diff
@@ -284,101 +256,98 @@ namespace rhost {
         }
 
         void create_blob(const message& msg) {
-            assert(msg.name == "?CreateBlob");
+            assert(!strcmp(msg.name(), "?CreateBlob"));
+            blob_id id = create_blob(msg.blob());
+            respond_to_message(msg, static_cast<double>(id));
+        }
 
-            if (msg.blob.size() != msg.blob_count) {
-                fatal_error("Incomplete blob creation message %s.", msg.id.c_str());
-            }
+        blob_id create_blob(blob&& blob) {
+            std::lock_guard<std::mutex> lock(blobs_mutex);
+            blob_id id = ++next_blob_id;
 
-            {
-                std::lock_guard<std::mutex> lock(received_blobs_mutex);
-                int blob_id = ++next_blob_id;
-                received_blobs[blob_id] = msg.blob;
-                respond_to_message(msg, static_cast<double>(blob_id));
+            // Check that it never overflows double mantissa, and provide immediate diagnostics if it happens.
+            if (id != blob_id(double(id))) {
```

Comment:
Out or curiosity, do you think 64-bit it would overflow? Or there is no portable type?